### PR TITLE
Add compression to object store

### PR DESCRIPTION
**Pull request by Jarema**
This commit also cleans up formatting of the spec.

### CODE REVIEW

File: adr/ADR-20.md [hunk 1]
```diff
@@ -176,31 +180,31 @@ type ObjectMeta struct {
...
-direct fields of ObjectInfo 
+Object Info is meta plus instance information.
+The fields in ObjectMeta are serialized in line as if they were
+direct fields of ObjectInfo
 
 ```go
 type ObjectInfo struct {
     ObjectMeta
-    
+
```
Comment (ripienaar) [inline adr/ADR-20.md:192]: not sure these spaces are desired, also below and above

File: adr/ADR-20.md [hunk 2]
```diff
@@ -98,6 +100,7 @@ type ObjectStoreConfig struct {
 	Storage     StorageType		// stream storate_type
 	Replicas    int			// stream replicas
 	Placement   Placement		// stream placement
+    Compression bool        // stream compression
```
Comment (ripienaar) [inline adr/ADR-20.md:103]: spacing here

**Review by ripienaar — CHANGES_REQUESTED** [review]
looks some unintended spacing changes crept in

But mainly this doesnt seem to describe what the compression option does? see recent addition to KV adr for sample verbiage etc

File: adr/ADR-20.md (hunk 2 above)
Comment (Jarema) [reply adr/ADR-20.md:103]: yeah, fixed. My editor defaulted to spaces here.

File: adr/ADR-20.md (hunk 1 above)
Comment (Jarema) [reply adr/ADR-20.md:192]: Those spaces were removed in this PR, not added (by markdown linter). I have no idea why you would want them here, but the docs renders the same.

Comment (ripienaar) [reply adr/ADR-20.md:192]: oh god, need more coffee sorry, thought you were adding them lol.

Comment (ripienaar) [reply adr/ADR-20.md:192]: Right I see whats happening, you're removing spaces fine - but I think the entire line should be removed.

**Comment by Jarema** [comment]
@ripienaar the verbiage was compied from KV ADR PR, but I realized it was improved as a follow up :).

Improved it a bit.
However in general, this ADR is not as detailed and fleshed out as the KV one. I will clean it up in separate PR to get it up to the standard.

File: adr/ADR-20.md (hunk 1 above)
Comment (Jarema) [reply adr/ADR-20.md]: Yeah, you're right, removed them entirely.
It was just linter cleaning things up. I didn't want to change more than required for this PR, as I will cleanup the ADR as follow up :).

File: adr/ADR-20.md [hunk 3]
```diff
@@ -98,6 +100,7 @@ type ObjectStoreConfig struct {
 	Storage     StorageType		// stream storate_type
 	Replicas    int			// stream replicas
 	Placement   Placement		// stream placement
+	Compression  bool		// stream compression, s2 or none
```
Comment (ripienaar) [inline adr/ADR-20.md:103]: its not `s2 or none`

Comment (ripienaar) [reply adr/ADR-20.md:103]: The missing verbiage I mention is something like ``` * If compression is requested in the configuration set compression to s2``` that was added to kv adr

**Review by ripienaar — CHANGES_REQUESTED** [review]

File: adr/ADR-20.md (hunk 3 above)
Comment (ripienaar) [reply adr/ADR-20.md]: Yeah when implementing KV devs were confused thats why it was added, so just pre-empting.

**Review by ripienaar — APPROVED** [review]
LGTM

**Pull request merged**